code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm

```diff
@@ -131,21 +143,40 @@
 
 /datum/eldritch_knowledge/final/rust_final
 	name = "Rustbringer's Oath"
-	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become more resillient overall."
+	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become extremely more resillient."
 	gain_text = "Champion of rust. Corruptor of steel. Fear the dark for the Rustbringer has come! Rusted Hills, CALL MY NAME!"
 	cost = 3
 	required_atoms = list(/mob/living/carbon/human)
 	route = PATH_RUST
+	var/list/conditional_immunities = list(TRAIT_STUNIMMUNE,TRAIT_SLEEPIMMUNE,TRAIT_PUSHIMMUNE,TRAIT_SHOCKIMMUNE,TRAIT_NOSLIPALL,TRAIT_RADIMMUNE,TRAIT_RESISTHIGHPRESSURE,TRAIT_RESISTLOWPRESSURE,TRAIT_RESISTCOLD,TRAIT_RESISTHEAT,TRAIT_PIERCEIMMUNE,TRAIT_BOMBIMMUNE,TRAIT_NOBREATH)
+	///if this is set to true then immunities are active, if false then they are not active, simple as.
+	var/delta = FALSE
 
 /datum/eldritch_knowledge/final/rust_final/on_finished_recipe(mob/living/user, list/atoms, loc)
 	var/mob/living/carbon/human/H = user
 	H.physiology.brute_mod *= 0.5
 	H.physiology.burn_mod *= 0.5
 	H.client?.give_award(/datum/award/achievement/misc/rust_ascension, H)
+	RegisterSignal(H,COMSIG_MOVABLE_MOVED,.proc/on_move)
 	priority_announce("$^@&#*$^@(#&$(@&#^$&#^@# Fear the decay, for the Rustbringer, [user.real_name] has ascended! None shall escape the corrosion! $^@&#*$^@(#&$(@&#^$&#^@#","#$^@&#*$^@(#&$(@&#^$&#^@#", ANNOUNCER_SPANOMALIES)
 	new /datum/rust_spread(loc)
 	return ..()
 
+/datum/eldritch_knowledge/final/rust_final/proc/on_move(mob/mover)
+	SIGNAL_HANDLER
+	var/expression = istype(get_turf(mover),/turf/open/floor/plating/rust)
```

Comment:
What is this name supposed to mean? Why not just `is_on_rust`?

Reply:
because the proc doesn't run when user is on rust, but rather when they move, expression for when the user *is* on rust is the first if statement.

Reply:
That still doesn't really explain the choice of the name "expression"--it's confusing.

Reply:
the variable just holds the return value of istype which is either TRUE or FALSE, making it a boolean expression, i could rename it to boolean or sth for clarity, but basically what the code does is it checks wether we are on a rust tile, if yes we then check the delta, if the delta points towards us not being on a rust tile before then we apply effects. If the expression is false but delta is true it de-applies the effects, in this case we check if we are *not* standing on a rust tile and if we *were* standing on it last time we checked.

Reply:
changed the names